src/turret.rs

```diff
@@ -60,6 +70,46 @@ pub struct SpawnTurret {
 
 impl Command for SpawnTurret {
     fn apply(self, world: &mut World) {
+        let mut state: SystemState<(
+            Query<(
+                &mut RandomDeterministic,
+                &mut crate::inventory::Inventory<buildings::ItemType>,
+            )>,
+            Query<&buildings::ItemType>,
+        )> = SystemState::new(world);
+
+        let mut new_item = || {
+            let (mut q_inventory, q_items) = state.get_mut(world);
+
+            let (mut rng, mut inventory) = q_inventory.single_mut();
+
+            let Some(first_item) = inventory.items.front().cloned() else {
+                return None;
+            };
+            let Ok(item_to_build) = q_items.get(first_item) else {
+                return None;
+            };
+            // TODO: check if we can build item_to_build (cooldown, space available, currency, ...)
+            // TODO: send an event if not possible.
+            // TODO: pay "price" ?
+            inventory.items.pop_front();
+
+            /*
+            let new_item = buildings::get_random_item(&mut rng);
+            let new_item = world.spawn(new_item).id();*/
+            Some((first_item, first_item))
+        };
+
+        let Some((item_built, new_item)) = new_item() else {
+            return;
+        };
+        // TODO: reuse that entity to merge it with turret entity ?
+        world.despawn(item_built);
+        let (mut q_inventory, q_items) = state.get_mut(world);
+        let (mut rng, mut inventory) = q_inventory.single_mut();
+
+        //inventory.items.push_back(new_item);
```

Comment:
I think that shouldn't be commented
```suggestion
        inventory.items.push_back(new_item);
```

Reply:
or we can feed the queue from another system, to have some kind of cooldown (every second, a new item appears in the queue, so you can't spam too much)